hoomd/md/mesh/potential.py

```diff
@@ -58,6 +58,17 @@ def _apply_typeparam_dict(self, cpp_obj, simulation):
                     f"For {type(self)} in TypeParameter {typeparam.name} "
                     f"{str(err)}")
 
+    @property
+    def all_params(self):
+        """Type parameter setter for all mesh types."""
+        raise RuntimeError("all_params can only be used as a setter.")
+        return []
+
+    @all_params.setter
+    def all_params(self, value):
+        for mt in self.mesh.types:
+            self.params[mt] = value
+
```

Comment:
`TypeParameter` objects support a default value `tp.default = ...` which will be used for all types not explicitly set. This also applies for types where one parameter was set but not another.